src/node_types/sequencer.rs

```diff
@@ -263,38 +263,61 @@ impl Sequencer {
         match operation {
             Operation::Add { id, value } | Operation::Revoke { id, value } => {
                 // verify that the hashchain already exists
-                self.db.get_hashchain(id).map_err(|e| {
+                let mut current_chain = self.db.get_hashchain(id).map_err(|e| {
                     DatabaseError::NotFoundError(format!("hashchain for ID {}: {}", id, e))
                 })?;
 
                 let mut tree = self.tree.lock().await;
                 let hashed_id = sha256_mod(id.as_bytes());
 
-                let node = match tree.find_leaf_by_label(&hashed_id) {
-                    Some(node) => node,
-                    None => {
-                        // TODO: before merging, change error type
-                        return Err(GeneralError::DecodingError(format!(
-                            "node with label {} not found in the tree",
-                            hashed_id
-                        ))
-                        .into());
-                    }
-                };
-                let new_index = match tree.find_node_index(&node) {
-                    Some(index) => index,
-                    None => {
-                        return Err(GeneralError::DecodingError(format!(
-                            "node with label {} not found in the tree, but has a hashchain entry",
-                            hashed_id
-                        ))
-                        .into());
-                    }
+                let node = tree.find_leaf_by_label(&hashed_id).ok_or_else(|| {
+                    // TODO: before merging, change error type
+                    GeneralError::DecodingError(format!(
+                        "node with label {} not found in the tree",
+                        hashed_id
+                    ))
+                })?;
+
+                let updated_node = Node::new_leaf(
+                    node.is_left_sibling(),
+                    hashed_id,
+                    sha256_mod(value.as_bytes()), // value shouldnt be hashed, right?
```

Comment:
value should be hashed in the node, just not in the TD, so sha256_mod here is correact

Reply:
ohh, but the value in the node is the latest value from the hashchain, so i need to change this anyways